internal/dag/policy_test.go

```diff
@@ -232,7 +232,7 @@ func TestTimeoutPolicy(t *testing.T) {
 				Idle: "900s",
 			},
 			want: TimeoutPolicy{
-				IdleTimeout: timeout.DurationSetting(900 * time.Second),
+				IdleStreamTimeout: timeout.DurationSetting(900 * time.Second),
```

Comment:
would be nice to add cases for invalid & valid idle connection timeouts here, for consistency

Reply:
Added tests for invalid, valid timeout and infinite.